lib/cpp/src/thrift/stdcxx.h

```diff
@@ -102,23 +108,71 @@ namespace apache { namespace thrift { namespace stdcxx {
   using ::boost::enable_shared_from_this;
   using ::boost::make_shared;
   using ::boost::scoped_ptr;
+  using ::boost::scoped_array;
   using ::boost::shared_ptr;
   using ::boost::static_pointer_cast;
   using ::boost::weak_ptr;
 
+#if (BOOST_VERSION >= 105700)
+  using ::boost::movelib::unique_ptr;
+#else
+  using ::boost::interprocess::unique_ptr;
+#endif
+
 #else
 
   using ::std::const_pointer_cast;
   using ::std::dynamic_pointer_cast;
   using ::std::enable_shared_from_this;
   using ::std::make_shared;
   template <typename T> using scoped_ptr = std::unique_ptr<T>;		// compiler must support template aliasing
+  template <typename T> using scoped_array = std::unique_ptr<T[]>;      // compiler must support template aliasing
   using ::std::shared_ptr;
   using ::std::static_pointer_cast;
   using ::std::weak_ptr;
+  using ::std::unique_ptr;
+
+#endif
+
+}}} // apache::thrift::stdcxx
+
+///////////////////////////////////////////////////////////////////
+//
+// Atomic
+//
+///////////////////////////////////////////////////////////////////
+
+#if __cplusplus < 201103L
```

Comment:
This is always true on all Microsoft compilers (2010 - 2017).  It isn't the best thing to be conditional on and be portable, unfortunately.